tests/tests_core/test_core_util.py

```diff
@@ -32,19 +37,30 @@ def test_rotation_matrix_z():
 
 
 def test_spherical_to_cartesian():
-    result = spherical_to_cartesian(np.array([0.5, np.pi / 4, -np.pi / 4]))
-    expected = np.array([0.25, -0.25, 0.35355339])
-    assert np.allclose(expected, result)
+    def sph_car(sph, car):
+        result = spherical_to_cartesian(sph)
+        assert np.allclose(car, result)
+        result = cartesian_to_spherical(result)
+        expect = cartesian_to_spherical(car)
+        assert np.allclose(result, expect)
+        result = spherical_to_cartesian(cartesian_to_spherical(car))
+        assert np.allclose(car, result)
+
+    sph_car(np.array([0.5, np.pi / 4, -np.pi / 4]), np.array([0.25, -0.25, 0.35355339]))
+
+    sph_car(
+        np.array([0.5, -np.pi / 4, np.pi / 4]), np.array([-0.25, -0.25, 0.35355339])
+    )
 
-    result = spherical_to_cartesian(np.array([0.5, -np.pi / 4, np.pi / 4]))
-    expected = np.array([-0.25, -0.25, 0.35355339])
-    assert np.allclose(expected, result)
+    sph_car(
+        np.array([[0.5, np.pi / 4, -np.pi / 4], [0.5, -np.pi / 4, np.pi / 4]]),
+        np.array([[0.25, -0.25, 0.35355339], [-0.25, -0.25, 0.35355339]]),
+    )
 
-    result = spherical_to_cartesian(
-        np.array([[0.5, np.pi / 4, -np.pi / 4], [0.5, -np.pi / 4, np.pi / 4]])
+    sph_car(
+        np.array([2.60564963, 1.75305207, 4.4458828]),
+        np.array([-0.674864797187, -2.472029259161, -0.472269863940]),
```

Comment:
I have no experience with pytest so I cannot give an opinion. But these particular values were chosen because they trigger corner cases due to the use of arctan2 and arccos. There are probably nicer ways to test the corner cases but I didn't have much time to think about them.

Reply:
In fact, if there was a way to sample the full range of values in terms of positive/negative directions, that would be even better.

Reply:
Right! I'll have a look at this and try a version with Hypothesis + pytest then